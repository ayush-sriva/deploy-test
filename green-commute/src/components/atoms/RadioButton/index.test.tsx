import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import RadioButtonComponent from '.';

describe('Atoms', () => {
  test('renders radio button', () => {
    render(<RadioButtonComponent label={'Yes'} />);
    const radioLabel: HTMLInputElement = screen.getByLabelText('Yes');
    expect(radioLabel.checked).toEqual(false);
    fireEvent.click(radioLabel);
    expect(radioLabel.checked).toEqual(true);
  });
});

describe('Atoms', () => {
  test('renders snapshot of radio button', () => {
    const tree = renderer.create(<RadioButtonComponent label='Yes' />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
