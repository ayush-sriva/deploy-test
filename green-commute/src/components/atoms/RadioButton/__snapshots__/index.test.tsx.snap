// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms renders snapshot of radio button 1`] = `
<label
  className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-spe08f-MuiFormControlLabel-root"
>
  <span
    className="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-wgff7p-MuiButtonBase-root-MuiRadio-root"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <input
      className="PrivateSwitchBase-input css-1m9pwf3"
      onChange={[Function]}
      type="radio"
      value="Yes"
    />
    <span
      className="css-16raxqw"
      fontSize="medium"
    />
  </span>
  <span
    className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
  >
    Yes
  </span>
</label>
`;
